force-app/main/default/classes/ContactTriggerHandler.cls

```diff
@@ -0,0 +1,71 @@
+public with sharing class ContactTriggerHandler {
+        
+    boolean callNewJob = false;
+    Id accountIdToBeUpdated;
+    string phoneToBeUpdated;
+
+    //Create validation for already existing primary contacts in accounts;
+    public void handleBeforeInsert(List<Contact> newTriggerContact){
+                
+        List<Contact> notAllowedContacts = new List<Contact>();
+        //Comparing primary contacts to be added to accounts that has primary contacts already.
+        //If matches, they will be included in a black list.        
+        for(Contact contact : newTriggerContact){
+            if(!isValidPrimaryContactForInsert(contact.AccountId)){
+				notAllowedContacts.add(contact);
+            }
+        }   
```

Comment:
Most probably this code will result in throwing an exception, since, for each contact, a SOQL query is made in isValidPrimaryContactForInsert method.